asset_stream_manager/base_command.cc

```diff
@@ -0,0 +1,71 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "asset_stream_manager/base_command.h"
+
+#include "lyra/lyra.hpp"
+
+namespace cdc_ft {
+
+BaseCommand::BaseCommand(std::string name, std::string help, int* exit_code)
+    : name_(name), help_(help), exit_code_(exit_code) {
+  assert(exit_code_);
+}
+
+BaseCommand::~BaseCommand() = default;
+
+void BaseCommand::Register(lyra::cli& cli) {
+  lyra::command cmd(name_,
+                    [this](const lyra::group& g) { this->CommandHandler(g); });
+  cmd.help(help_);
+  cmd.add_argument(lyra::help(show_help_));
+
+  RegisterCommandLineFlags(cmd);
+
+  // Workaround for Lyra treating --unknown_flags as positional argument.
```

Comment:
It might be a good idea to add to the asset-stream-manager main a comment, for example: no positional arguments are supported. By adding a new option, a new flag should be introduced as well.